mm/2s2h/Rando/Logic/Regions/South.cpp

```diff
@@ -5,17 +5,42 @@
 
 using namespace Rando::Logic;
 
-// clang-format off
-static RegisterShipInitFunc initFunc([]() {
-    Regions[RR_DEKU_KINGS_CHAMBER_HOLDING_CELL] = RandoRegion{ .name = "Holding Cell", .sceneId = SCENE_DEKU_KING,
+inline bool CanGetPastBigOcto() {
+    // Boat ride. This takes Link to a landing in the next region and thus has no additional requirements.
+    if (RANDO_EVENTS[RE_SOUTHERN_SWAMP_RIDE_BOAT]) {
+        return true;
+    }
+    /*
+     * Kill the Big Octo. This means the water is still in a poison state, so swimming is not logically considered due
+     * to the damage boost method not being logically considered. This piece may need expanded as tricks and glitches
+     * are logically introduced.
+     */
+    if (RANDO_EVENTS[RE_SOUTHERN_SWAMP_KILL_OCTOROK] && CAN_BE_DEKU) {
+        return true;
+    }
+    /*
+     * The water is clear in this state, so swimming comes into question. Either Link uses the swim ability, he hops
+     * over the water as Deku, or he uses one of the forms tall enough to walk through the water. This may also be
+     * expanded if FD is ever logically considered.
+     */
+    if (RANDO_EVENTS[RE_CLEARED_WOODFALL_TEMPLE] &&
+        (CAN_USE_ABILITY(SWIM) || CAN_BE_DEKU || CAN_BE_ZORA || CAN_BE_GORON)) {
+        return true;
+    }
+    return false;
+}
+
```

Comment:
Sorry, I meant to for `CanGetPastBigOcto` to go before the clang format line, but not to remove it. Can you walk back the `e09482f` clang format commit and add this comment back? What I want is for `CanGetPastBigOcto` to be formatted, but not the rest of the file.
```suggestion

// clang-format off
```

Reply:
I gotcha now, fixed.